packages/app/src/cli/services/function/ui/components/Replay.tsx

```diff
@@ -0,0 +1,168 @@
+import {setupExtensionWatcherForReplay} from './hooks/setup-extension-watcher-for-replay.js'
+import {FunctionRunData} from '../../replay.js'
+import {ExtensionInstance} from '../../../../models/extensions/extension-instance.js'
+import {FunctionConfigType} from '../../../../models/extensions/specifications/function.js'
+import {AppInterface} from '../../../../models/app/app.js'
+import {prettyPrintJsonIfPossible} from '../../../app-logs/utils.js'
+import figures from '@shopify/cli-kit/node/figures'
+import {AbortController} from '@shopify/cli-kit/node/abort'
+import React, {FunctionComponent} from 'react'
+import {Box, Text, Static} from '@shopify/cli-kit/node/ink'
+
+export interface ReplayProps {
+  selectedRun: FunctionRunData
+  abortController: AbortController
+  app: AppInterface
+  extension: ExtensionInstance<FunctionConfigType>
+}
+
+interface FunctionRun {
+  type: 'functionRun'
+  input: string
+  output: string
+  logs: string
+  name: string
+  size: number
+  memory_usage: number
+  instructions: number
+}
+
+interface SystemMessage {
+  type: 'systemMessage'
+  message: string
+}
+
+type ReplayLog = FunctionRun | SystemMessage
+
+const Replay: FunctionComponent<ReplayProps> = ({selectedRun, abortController, app, extension}) => {
+  const {logs, isAborted, canUseShortcuts, statusMessage, recentFunctionRuns, error} = setupExtensionWatcherForReplay({
+    selectedRun,
+    abortController,
+    app,
+    extension,
+  })
+
+  return (
+    <>
+      {/* Scrolling upper section */}
+      <Static items={logs}>
+        {(log, index) => {
+          return (
+            <Box key={`replayOutputScrollerLog${index}`} flexDirection="column">
+              <ReplayLog log={log} />
+            </Box>
+          )
+        }}
+      </Static>
+      {/* Bottom Bar */}
+      {/* eslint-disable-next-line no-negated-condition */}
+      {!isAborted ? (
+        <Box
+          marginY={1}
+          paddingTop={1}
+          flexDirection="column"
+          flexGrow={1}
+          borderStyle="single"
+          borderBottom={false}
+          borderLeft={false}
+          borderRight={false}
+          borderTop
+        >
+          {canUseShortcuts ? (
+            <Box flexDirection="column">
+              <Box flexDirection="row">
+                <Text>
+                  {figures.pointerSmall} {statusMessage}
+                </Text>
+              </Box>
+              <StatsDisplay recentFunctionRuns={recentFunctionRuns} />
+              <Text>
+                {figures.pointerSmall} Press <Text bold>q</Text> {figures.lineVertical} quit
+              </Text>
+            </Box>
+          ) : null}
+          {error ? <Text color="red">{error}</Text> : null}
+        </Box>
+      ) : null}
+    </>
+  )
+}
+
+function InputDisplay({input}: {input: string}) {
+  return (
+    <Box flexDirection="column">
+      <Text color="black" backgroundColor="yellow">
+        {'\n\n            Input            \n'}
```

Comment:
Would it make sense for this to be a component that just centres some text? It looks like it's reused in a bunch of these components.